### PR TITLE
Migrate catalog to FCB format

### PR DESCRIPTION
**Pull request by akram**
**What type of PR is this?**
/kind enhancement

**What does this PR do / why we need it**:
The opm `sqlite` catalog format is now deprecated in favor of flat based catalog.

**Have you updated the necessary documentation?**

* [ ] Documentation update is required by this PR.
* [ ] Documentation has been updated.

**Which issue(s) this PR fixes**:

Fixes #?

**Test acceptance criteria**:

* [ ] Unit Test
* [ ] E2E Test

**How to test changes / Special notes to the reviewer**:
A new catalog image can be generated by using:
```
make bundle-build bundle-push catalog-build catalog-push  IMAGE=quay.io/akram/gitops-backend-operator
```
replace `IMAGE` with a repository where you have permissions.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -241,7 +241,14 @@ endif
 # https://github.com/operator-framework/community-operators/blob/7f1438c/docs/packaging-operator.md#updating-your-existing-operator
 .PHONY: catalog-build
 catalog-build: opm ## Build a catalog image.
-	$(OPM) index add --container-tool docker --mode semver --tag $(CATALOG_IMG) --bundles $(BUNDLE_IMGS) $(FROM_INDEX_OPT)
+	rm -fr catalog/ catalog.Dockerfile
```
Comment (iam-veeramalla) [inline Makefile:244]: I see there is a space in the command. 

Is it intended ?

File: Makefile [hunk 2]
```diff
@@ -241,7 +241,14 @@ endif
...
+	mkdir -p catalog/
+	$(OPM) generate dockerfile catalog
+	$(OPM) init gitops-operator --default-channel=preview --description=./README.md --icon=./docs/assets/gitops-operator.svg --output yaml > catalog/operator.yaml
```
Comment (iam-veeramalla) [inline Makefile:247]: --default-channel=preview ?

**Comment by iam-veeramalla** [comment]
> **What type of PR is this?** /kind enhancement
> 
> **What does this PR do / why we need it**: The opm `sqlite` catalog format is now deprecated in favor of flat based catalog.
> 
> **Have you updated the necessary documentation?**
> 
>     * [ ]  Documentation update is required by this PR.
> 
>     * [ ]  Documentation has been updated.
> 
> 
> **Which issue(s) this PR fixes**:
> 
> Fixes #?
> 
> **Test acceptance criteria**:
> 
>     * [ ]  Unit Test
> 
>     * [ ]  E2E Test
> 
> 
> **How to test changes / Special notes to the reviewer**: A new catalog image can be generated by using:
> 
> ```
> make bundle-build bundle-push catalog-build catalog-push  IMAGE=quay.io/akram/gitops-backend-operator
> ```
> 
> replace `IMAGE` with a repository where you have permissions.

Any reference document for the below comment ?
> The opm `sqlite` catalog format is now deprecated in favor of flat based catalog.

**Comment by sbose78** [comment]
Did you mean, "FBC format" ? :-)

**Comment by varshab1210** [comment]
Closing this, please reopen if required

**Pull request closed without merging**